### PR TITLE
Added support to use Managed Identity to connect to Azure SQL 

### PR DESCRIPTION
**Pull request by pega-sagas1**

### CODE REVIEW

File: charts/pega/templates/_pega-deployment.tpl [hunk 1]
```diff
@@ -48,6 +48,9 @@ spec:
     metadata:
       labels:
         app: {{ .name }}
+        {{- if .root.Values.global.jdbc.podbinding }}
+        aadpodidbinding: {{ .root.Values.global.jdbc.podbinding }}
+        {{- end  }}
```
Comment (kishorv10) [inline charts/pega/templates/_pega-deployment.tpl:53]: Sumit, I don't think we need to expose the option for taking `addpodidbinding` or explicitly provide the label in deployment. The provision to add extra labels to the deployment is sufficient. See line # 54, labels can be added using `podLabels` option. Clients can use this configuration to add the required labels. We can update the documentation to use this option for adding more labels

File: charts/pega/values-large.yaml [hunk 2]
```diff
@@ -59,10 +59,16 @@ global:
...
+    # custom Authentication using Managed Identity for `action=INSTALL` should be passed here:
+    # For example: Authentication=ActiveDirectoryMSI;msiClientId=1234;
     connectionProperties: ""
     rulesSchema: "YOUR_RULES_SCHEMA"
     dataSchema: "YOUR_DATA_SCHEMA"
     customerDataSchema: ""
+    # As an authentication alternative, you can configure an AAD Pod Identity to manage
+    # authentication access for VMSS of your cluster to Azure SQL database.
+    # PodBinding adds `aadpodidbinding` label to pod for AAD Pod Identity discovery.
+    podbinding: ""
```
Comment (kishorv10) [inline charts/pega/values-large.yaml:71]: Lets not add values.yaml option for this configuration

File: charts/pega/charts/installer/templates/_pega-installer-job.tpl [hunk 3]
```diff
@@ -29,6 +29,9 @@ spec:
     metadata:
       labels:
         installer-job: {{ .name }}
+        {{- if .root.Values.global.jdbc.podbinding }}
+        aadpodidbinding: {{ .root.Values.global.jdbc.podbinding }}
+        {{- end  }}
```
Comment (kishorv10) [inline charts/pega/charts/installer/templates/_pega-installer-job.tpl:34]: I guess for the installer there is no option to add labels from values. Probably its a good candidate to expose the option.

File: docs/Deploying-Pega-on-AKS.md [hunk 4]
```diff
@@ -515,7 +516,11 @@ helm inspect values pega/pega > <local filepath>/aks-demo/pega.yaml
...
 
-3. Save the file.
+4. Save the file.
+
+   > Note: To connect AKS with Azure SQL using managed Identity, you must include the `Authentication` type as well as `msiClientId` if using `user assigned managed identity` under `jdbc.connectionProperties` for action=`INSTALL`.
```
Comment (taz-pega-work) [inline docs/Deploying-Pega-on-AKS.md:521]: one little tweak>
> Note: To connect AKS with an Azure SQL database using managed Identity, you must include the `Authentication` type as well as `msiClientId` if using `user assigned managed identity` under `jdbc.connectionProperties` for action=`INSTALL`.  
example:

**Review by taz-pega-work — COMMENTED** [review]
some minor suggestions...hopefully these will help

**Review by taz-pega-work — APPROVED** [review]
looks good. thank you!

**Comment by kishorv10** [comment]
Can you add issue id to the PR title?

File: charts/pega/values.yaml [hunk 5]
```diff
@@ -59,10 +59,15 @@ global:
...
     # For example: connectionProperties=user=usr;password=pwd;
+    # custom Authentication using Managed Identity for `action=INSTALL` should be passed here:
+    # For example: Authentication=ActiveDirectoryMSI;msiClientId=1234;
     connectionProperties: ""
     rulesSchema: "YOUR_RULES_SCHEMA"
     dataSchema: "YOUR_DATA_SCHEMA"
     customerDataSchema: ""
+    # As an authentication alternative, you can configure an AAD Pod Identity to manage
+    # authentication access for VMSS of your cluster to Azure SQL database.
+    # use node.podLabels to pass `aadpodidbinding` label to pod for AAD Pod Identity discovery.
```
[the remark this answers is not in the document]
Comment (pega-sagas1) [reply charts/pega/values.yaml:70]: moved to the AKS documention. Can you please review @kishorv10 @taz-pega-work

File: docs/Deploying-Pega-on-AKS.md [hunk 6]
```diff
@@ -521,6 +521,18 @@ helm inspect values pega/pega > <local filepath>/aks-demo/pega.yaml
 example:
    `connectionProperties: "selectMethod=cursor;Authentication=ActiveDirectoryMSI;msiClientId=<clientId>;"`
 
+    As an authentication alternative, you can configure an AAD Pod Identity to manage authentication access for VMSS of your cluster to Azure SQL database. use
```
Comment (taz-pega-work) [inline docs/Deploying-Pega-on-AKS.md:524]: Fix context (this is pretty complex stuff) and then maybe confirm that it is clear how you add the `aadpodidbinding: <label>` label: there are no quotes here but don't all of the parameters require text to be parsed using double quotes?

As an authentication alternative, you can configure an AAD Pod Identity to manage authentication access for VMSS of your cluster to Azure SQL database. Use the "web" `tier.podLabels` parameter to pass a `aadpodidbinding` label to your pod to enable it to discover your AAD Pod Identity.

**Review by taz-pega-work — COMMENTED** [review]
close but please consider these review comments to clarify the context a bit.

**Review by taz-pega-work — APPROVED** [review]
thanks!!

**Review by kishorv10 — APPROVED** [review]

File: charts/pega/values-large.yaml [hunk 7]
```diff
@@ -59,8 +59,7 @@ global:
     # CUSTOM CONNECTION PROPERTIES
     # Add a list of ; delimited connections properties. The list must end with ;
```
Comment (taz-pega-work) [inline charts/pega/values-large.yaml:60]: Let's try this slight edit....is it `` or ""?
Add a list of semi-colon-delimited required connection properties to pass as part of the connection. 
The list must end with ;. For example, to set a custom authentication using Azure Managed Identity
Set  `connectionProperties: "Authentication=ActiveDirectoryMSI;msiClientId=1234;"`

File: charts/pega/values-minimal.yaml [hunk 8]
```diff
@@ -55,8 +55,7 @@ global:
     # CUSTOM CONNECTION PROPERTIES
     # Add a list of ; delimited connections properties. The list must end with ;
```
Comment (taz-pega-work) [inline charts/pega/values-minimal.yaml]: same as above

File: charts/pega/values.yaml [hunk 9]
```diff
@@ -59,8 +59,7 @@ global:
     # CUSTOM CONNECTION PROPERTIES
```
Comment (taz-pega-work) [inline charts/pega/values.yaml:59]: same as above

**Review by taz-pega-work — COMMENTED** [review]
added a few comments

**Review by taz-pega-work — COMMENTED** [review]
looks good! ;-)

**Comment by kishorv10** [comment]
#401

**Comment by kishorv10** [comment]
#402

**Pull request merged**